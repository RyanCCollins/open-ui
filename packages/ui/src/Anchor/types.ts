import { Props } from './';

export { Props };
