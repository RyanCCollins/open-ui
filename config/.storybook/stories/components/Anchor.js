import React from 'react';
import { storiesOf, action } from '@kadira/storybook';
import { Anchor } from 'ui';
import { 
  withKnobs,
  string,
  color
 } from '@kadira/storybook-addon-knobs';

 const stories = storiesOf('Anchor', module);
 stories.addDecorator(withKnobs);

stories
  .addWithInfo(
    'default anchor',
    `
      Properties:

      \`path: {string} \`

        Navigation path
      
      \`label: {string}\`

        Label seen on the Anchor component
      
      \`onClick: {function ()}\`

        onClick handler for the Anchor.
      
      Usage:

      ~~~js
        <Anchor
          onClick={this.handleClick}
          color='cornflowerblue'
          path='/about'
        />
      ~~~
    `,
     () =>
      <Anchor
        onClick={action('clicked the anchor component')}
        path="/docs"
      >
        Default Anchor
      </Anchor>
    ,
    {
      inline: true,
      propTables: false,
      source: false,
    }
  )
  .addWithInfo(
    'cornflower blue anchor',
    `
      Properties: 

      All the properties from the default \`<Anchor />\` component are available 
      
      additionally you can also pass in a \`"color"\` prop: 
      
        \`color: {string}\` 
      
      Usage:

      ~~~js
        <Anchor
          color='cornflowerblue'
          path='/about'
        />
      ~~~
    `,
     () =>
      <Anchor
        color={color('color','cornflowerblue')}
        path="/docs"
      >
        Cornflower Blue Anchor
      </Anchor>
    ,
    {
      inline: true,
      propTables: false,
      source: false,
    }
  );
