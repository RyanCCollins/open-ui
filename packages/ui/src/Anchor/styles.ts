import styled from 'styled-components';

export const AnchorComponent = styled.a`
  color: ${(props) => props.color};
  font-size: 1.1875rem;
  line-height: 24px;
  font-weight: 400;
`;
