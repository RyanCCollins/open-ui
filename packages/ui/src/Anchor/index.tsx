import * as React from 'react';
import { withRouter } from 'react-router';
import { AnchorComponent } from './styles';

const MethodTypePush = 'push';
const MethodTypeReplace = 'replace';
type Method = 'push' | 'replace';
export interface Props extends React.Props<Anchor> {
  path?: string;
  href?: string;
  label?: string;
  color?: string;
  method?: Method;
  router?: {
    push: Function,
    replace: Function,
    createPath: Function,
  }
}

class Anchor extends React.Component<Props, undefined> {
  public static defaultProps: Props = {
    color: '#007acc',
    method: MethodTypePush,
  };

  private handleClick = (e: React.MouseEvent<HTMLAnchorElement>): void => {
    const { method, path, router } = this.props;
    if (path) {
      e.preventDefault();
      if (method === MethodTypePush) {
        router.push(path);
      } else if (method === MethodTypeReplace) {
        router.replace(path);
      }
    }
  }

  public render() {
    const {
      label,
      children,
      color,
      href,
    } = this.props;

    return (
      <AnchorComponent href={href} color={color} onClick={this.handleClick}>
        {label || children}
      </AnchorComponent>
    );
  }
}

export default withRouter(Anchor);
