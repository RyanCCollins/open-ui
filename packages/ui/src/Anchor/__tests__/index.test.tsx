import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import * as React from 'react';
import Anchor from '../';
import pathProps, { hrefProps } from './__mocks__/anchorMocks.mock';

describe('<Anchor />', () => {
  it('should render with default props', () => {
    const wrapper = shallow(
      <Anchor />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('should render path props', () => {
    const wrapper = shallow(
      <Anchor {...pathProps} />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('should render href props', () => {
    const wrapper = shallow(
      <Anchor {...hrefProps} />,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
  it('should render with children', () => {
    const wrapper = shallow(
      <Anchor path="/about">
        About ðŸ‘¶
      </Anchor>,
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
