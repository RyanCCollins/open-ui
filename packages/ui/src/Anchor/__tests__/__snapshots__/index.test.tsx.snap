// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Anchor /> should render href props 1`] = `
<Anchor
  color="#fff"
  href="https://github.com/RyanCCollins/scalable-react-ts-boilerplate"
  label="Scalable React"
  method="push"
/>
`;

exports[`<Anchor /> should render path props 1`] = `
<Anchor
  color="#fff"
  label="Docs"
  method="push"
  path="/docs"
/>
`;

exports[`<Anchor /> should render with children 1`] = `
<Anchor
  color="#007acc"
  method="push"
  path="/about"
>
  About ðŸ‘¶
</Anchor>
`;

exports[`<Anchor /> should render with default props 1`] = `
<Anchor
  color="#007acc"
  method="push"
/>
`;
